libnymea-core/devices/devicemanagerimplementation.cpp

```diff
@@ -1148,6 +1148,13 @@ void DeviceManagerImplementation::loadConfiguredDevices()
             }
         }
         DeviceUtils::verifyParams(deviceClass.settingsTypes(), deviceSettings);
+        // Make sure all settings are around. if they aren't initialize with default values
+        foreach (const ParamType &settingsType, deviceClass.settingsTypes()) {
+            if (!deviceSettings.hasParam(settingsType.id())) {
+                deviceSettings.append(Param(settingsType.id(), settingsType.defaultValue().isValid() ? settingsType.defaultValue() : ""));
+            }
+        }
+
```

Comment:
Here we might have the same issue as with the device params. Will the data type of the QVariant stored and loaded properly?

Reply:
The actual loading and storing happens above this in code that existed before already. This just verifies that all of the settings have a value. E.g. if a plugin adds a new setting in an update, the settings might not have it yet and this will load the defaultValue.